task/gimlet-inspector/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! The Gimlet Inspector provides a deliberately limited set of IPCs over the
+//! network, for extracting diagnostic data from a live system. This is intended
+//! to supplement the more general `dump_agent`.
+
+#![no_std]
+#![no_main]
+
+use drv_gimlet_seq_api::{SeqError, Sequencer};
+use gimlet_inspector_protocol::{
+    QueryV0, Request, SequencerRegistersResponseV0,
+};
+use hubpack::SerializedSize;
+use task_net_api::*;
+use userlib::*;
+
+task_slot!(NET, net);
+task_slot!(SEQ, seq);
+
+// Increase this if a new message requires a larger trailer.
+const MAX_REQ_TRAILER_SIZE: usize = 0;
```

Comment:
Should this be in `gimlet_inspector_protocol` instead, since that's the canonical definition of requests?

Reply:
Sure.

Reply:
Factored out and rejiggered.